src/trinerdi/data-structures/lazy_segtree.cpp

```diff
@@ -0,0 +1,56 @@
+/**
+ * Name: Lazy segment tree
+ * Author: Vaclav Volhejn
+ * Description: Lazy minimum segment tree supporting range updates
+ *  and queries. Exclusive right bounds.
+ * Time: $O(\log N)$ per update/query
+ */
+#include "../base.hpp"
+
+struct Segtree {
+    const ll INF = 1e18; // Neutral element of min
+    int l, r; // Exclusive righ
+    ll val = 0, lazy = 0;
+    Segtree *lson = NULL, *rson = NULL;
+
+    Segtree (int _l, int _r) : l(_l), r(_r) {}
+    ~Segtree() { delete lson; delete rson; }
+    
+    void unlazy() {
+        val += lazy;
+        if (l == r) return;
+        if(lson == NULL) {
```

Comment:
I don't know about C++ conventions, but in C, `if (!lson)` is the way to do it. (Readability may be an argument, but that isn't our main aim anyway.)